searx/searx
# Add Bandcamp search engine.

![Bandcamp engine](https://user-images.githubusercontent.com/25624882/103235055-2213eb80-490f-11eb-8333-8747fdeba25c.png)
## What does this PR do?
It adds [Bandcamp](https://bandcamp.com) as a search engine to Searx. I mostly modeled my engine around the existing SoundCloud engine and looked to the Bing engine for some XPath help. It supports the parsing of URLs, titles, content, published dates, embeds, and thumbnails from Bandcamp albums, tracks, and singles.

## Why is this change important?
Bandcamp is a large music platform and I was surprised to see that no one had created an engine for it yet,

## How to test this PR locally?
The engine is completely compatible with `make test`. If you want to see the engine in action:
* `make run`
* search for `!bc zelda & chill` under the music category to test album support
* search for `!bc prototype remix tall black guy` under the music category to test track/single support

## Author's checklist
I could make the content field meatier by adding tags and album length, but I wanted to keep my first commit simple.

P.S.: `chore: Added self to README.` should actually be `chore: Added self to AUTHORS.rst.` My mistake.
